packages/giselle-engine/src/core/generations/telemetry.ts

```diff
@@ -0,0 +1,43 @@
+import type { AnthropicProviderOptions } from "@ai-sdk/anthropic";
+import type { LanguageModel } from "@giselle-sdk/language-model";
+import type { ToolSet } from "ai";
+
+type TelemetryTag =
+	| "web-search"
+	| "search-grounding"
+	| "reasoning"
+	| "thinking";
+
+export function generateTelemetryTags(args: {
+	provider: string;
+	languageModel: LanguageModel;
+	toolSet: ToolSet;
+	configurations: Record<string, unknown>;
+	providerOptions?: {
+		anthropic?: AnthropicProviderOptions;
+	};
+}): TelemetryTag[] {
+	const tags: TelemetryTag[] = [];
+
+	// OpenAI Web Search
+	if (args.provider === "openai" && args.toolSet.openaiWebSearch) {
+		tags.push("web-search");
+	}
+
+	// Google Search Grounding
+	if (args.provider === "google" && args.configurations.searchGrounding) {
+		tags.push("search-grounding");
```

Comment:
```suggestion
		tags.push("web-search");
```

Reply:
IMO:

Since we define TelemetryTag ourselves, I thought it would be easier to understand if we grouped it with "web-search".

Reply:
It might be good to tag both our grouped definitions and the provider definitions. Like `web-search`, `google:search-grounding`.

Reply:
It seems like a good idea to add a namespace and define two tags. 👍

Reply:
> It seems like a good idea to add a namespace and define two tags

Thanks, I'll try to do that!

Reply:
fixed: https://github.com/giselles-ai/giselle/pull/816/commits/aa278b1052a6da889eda82d7f90b54e90f255f4b